pyfr/plugins/base.py

```diff
@@ -167,3 +169,90 @@ def __init__(self, intg, *args, **kwargs):
 
             if not isinstance(eidxs, slice):
                 self._ele_region_data[f'{etype}_idxs'] = eidxs
+
+
+class ElementMixin:
+    def __init__(self, intg, *args, **kwargs):
+        super().__init__(intg, *args, **kwargs)
+
+        self._surf_quad = {}
+
+    def surf_quad(self, itype, proj, *args):
+        try:
+            ppts, pwts = self._surf_quad[itype, proj]
+        except KeyError:
+            # Obtain quadrature info
+            rname = self.cfg.get(f'solver-interfaces-{itype}', 'flux-pts')
+
+            # Quadrature rule (default to that of the solution points)
+            qrule = self.cfg.get(self.cfgsect, f'quad-pts-{itype}', rname)
+            try:
+                qdeg = self.cfg.getint(self.cfgsect, f'quad-deg-{itype}')
+            except NoOptionError:
+                qdeg = self.cfg.getint(self.cfgsect, 'quad-deg')
+
+            q = get_quadrule(itype, qrule, qdeg=qdeg, flags='s')
+            ppts, pwts = self._proj_pts(proj, q.pts), q.wts
+            self._surf_quad[itype, proj] = (ppts, pwts)
+
+        return ppts, pwts
+
+    @staticmethod
+    def _proj_pts(projector, pts):
+        pts = np.atleast_2d(pts.T)
```

Comment:
Is this check needed in this context?

Reply:
Just following up here.

Reply:
I think it's a reasonable check in case someone tries to pass 1D quad points.

Reply:
It is not really a check.  It is to add support for 1D projections, which I do not think make sense here (we want to get an exception).